### PR TITLE
feature: Added collector for Alibaba cloud

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

**Review by dipsubha06 — APPROVED** [review]

File: collectors/alibaba/collector.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+            var callRegions = regions[serviceLower];
+            var requestOption = { method: callObj.method || 'POST' };
+
+            async.eachLimit(callRegions, helpers.MAX_REGIONS_AT_A_TIME, function(region, regionCb) {
+                if (settings.skip_regions &&
+                    settings.skip_regions.indexOf(region) > -1 &&
+                    globalServices.indexOf(service) === -1) return regionCb();
+                if (!collection[serviceLower][callKey][region]) collection[serviceLower][callKey][region] = {};
+
+                var LocalAlibabaConfig = JSON.parse(JSON.stringify(AlibabaConfig));
```
Comment (giorod3) [inline collectors/alibaba/collector.js:148]:
```suggestion
                let LocalAlibabaConfig = JSON.parse(JSON.stringify(AlibabaConfig));
```

File: collectors/alibaba/collector.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+
+                var endpoint = `https://${serviceLower}.aliyuncs.com`;
+                LocalAlibabaConfig['endpoint'] = endpoint;
```
Comment (giorod3) [inline collectors/alibaba/collector.js:151]: do we use the variable endpoint anywhere else? if not can we just assign it directly

File: collectors/alibaba/collector.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                LocalAlibabaConfig['apiVersion'] = callObj.apiVersion || apiVersion;
+                var client = new alicloud(LocalAlibabaConfig);
+                var paginating = false;
+                var pageNumber = 1;
```
Comment (giorod3) [inline collectors/alibaba/collector.js:155]:
```suggestion
                let pageNumber = 1;
```

File: collectors/alibaba/collector.js [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+                        setTimeout(function() {
+                            regionCb();
+                        }, callObj.rateLimit);
+                    } else {
+                        regionCb();
+                    }
+                };
+
+                function execute(nextTokens, paginateParams) {
+                    var localParams = JSON.parse(JSON.stringify(callObj.params || {}));
```
Comment (giorod3) [inline collectors/alibaba/collector.js:201]:
```suggestion
                    let localParams = JSON.parse(JSON.stringify(callObj.params || {}));
```

File: collectors/alibaba/collector.js [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+                    if (callObj.rateLimit) {
+                        setTimeout(function() {
+                            regionCb();
+                        }, callObj.rateLimit);
+                    } else {
+                        regionCb();
+                    }
+                };
+
+                function execute(nextTokens, paginateParams) {
```
Comment (giorod3) [inline collectors/alibaba/collector.js:200]:
```suggestion
                function execute(nextToken, paginateParams) {
```

File: collectors/alibaba/collector.js [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+                                }
+                            };
+
+                            var execute = function() {  
+                                client.request(callKey, params, requestOption).then((result) => {
+                                    requestCb(null, result);
+                                }, (err) => {
+                                    requestCb(err);
+                                });
+                            };
```
Comment (giorod3) [inline collectors/alibaba/collector.js:287]: we should have paginate logic here as well

File: collectors/alibaba/collector.js [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+    "api_calls": ["ECS:DescribeInstances", "VPC:DescribeFlowLogs"]
+}
+- callback: Function to call when the collection is complete
+*********************/
+
+var alicloud = require('@alicloud/pop-core');
+var async = require('async');
+var helpers = require(__dirname + '/../../helpers/alibaba');
+
+var apiVersion = '2015-05-01';
```
Comment (giorod3) [inline collectors/alibaba/collector.js:23]: remove

File: index.js [hunk 8]
```diff
@@ -122,29 +127,29 @@ function checkRequiredKeys(obj, keys) {
     });
 }
 
-if (config.credentials.aws.credential_file) {
+if (config.credentials.aws.credential_file && (!settings.cloud || (settings.cloud == 'aws'))) {
     cloudConfig = loadHelperFile(config.credentials.aws.credential_file);
```
Comment (giorod3) [inline index.js:131]: add settings.cloud here

File: index.js [hunk 9]
```diff
@@ -122,29 +127,29 @@ function checkRequiredKeys(obj, keys) {
...
     if (!cloudConfig || !cloudConfig.accessKeyId || !cloudConfig.secretAccessKey) {
         console.error('ERROR: AWS credential file does not have accessKeyId or secretAccessKey properties');
         process.exit(1);
     }
-} else if (config.credentials.aws.access_key) {
+} else if (config.credentials.aws.access_key && (!settings.cloud || (settings.cloud == 'aws'))) {
     checkRequiredKeys(config.credentials.aws, ['secret_access_key']);
```
Comment (giorod3) [inline index.js:137]: add settings.cloud here

**Review by giorod3 — APPROVED** [review]

**Pull request merged**